src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -450,8 +451,10 @@ private unsafe void RejectUnknownVersion(byte* version, int length)
             => throw GetInvalidRequestException(RequestRejectionReason.UnrecognizedHTTPVersion, version, length);
 
         [MethodImpl(MethodImplOptions.NoInlining)]
+#pragma warning disable CS0618 // Type or member is obsolete
         private unsafe BadHttpRequestException GetInvalidRequestException(RequestRejectionReason reason, byte* detail, int length)
```

Comment:
Use base type / using statement?

Reply:
Callers of this method throw the returned exception so I think it needs to be the old obsoleted version?

Reply:
Only the statement `new BadHttpRequestException` needs to specifically use the old type. All other fields, parameters, and return values can use the base type, they're just passing it through.

Reply:
Learnt something here. No problem, fixing.